### PR TITLE
Upgrade react-scripts to latest

### PR DESCRIPTION
**Pull request by darrenvong**
Since we recently upgraded TypeScript to the latest, may as well do the same for `react-scripts`. It also updates Jest to the latest version so probably will bring along some nice bug fixes too. The joy of a relatively new project when we can still upgrade things easily :smile:

### CODE REVIEW

File: src/components/Alert.tsx [hunk 1]
```diff
@@ -11,17 +14,17 @@ type AlertProps = MuiAlertProps & {
   severity: MuiAlertProps["severity"];
 };
 
-export default (props: AlertProps) => {
+export default function Alert(props: AlertProps) {
```
Comment (darrenvong) [inline src/components/Alert.tsx:17]: `react-scripts` introduced a new `import/no-anonymous-default-export` rule which meant these default exports without a function name were failing.

File: src/features/api.ts [hunk 2]
```diff
@@ -31,28 +31,23 @@ const opts = {
 
 // There seems to be an error in the `opts` parameter's type, so have to ignore that line.
 
-// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
```
Comment (darrenvong) [inline src/features/api.ts:34]: This is apparently not a rule anymore in the newer version of ESLint :thinking: No complaints here as this means fewer lines of code to use `@ts-ignore` for this exceptional case.

File: tsconfig.json [hunk 3]
```diff
@@ -14,12 +10,11 @@
     "forceConsistentCasingInFileNames": true,
     "module": "esnext",
     "moduleResolution": "node",
+    "noFallthroughCasesInSwitch": true,
```
Comment (darrenvong) [inline tsconfig.json:13]: See https://stackoverflow.com/questions/64115884/error-when-updating-create-react-app-to-4-0-with-typescript-template. It's the "recommended" value anyway so no harm done setting it explicitly.

Comment (amdwit) [reply tsconfig.json:13]: I don't like switch anyway. I prefer some if/else chaining or putting the cases in a mapping.

**Review by amdwit — APPROVED** [review]
Nice to be up to date indeed!

**Comment by lucaslcode** [comment]
Running `yarn` and `yarn start` on this branch, I get this error:

```
./node_modules/@pmmmwh/react-refresh-webpack-plugin/client/ReactRefreshEntry.js
Error: [BABEL] /home/lucas/couchers-react/node_modules/@pmmmwh/react-refresh-webpack-plugin/client/ReactRefreshEntry.js: Cannot find module 'levenary' (While processing: "/home/lucas/couchers-react/node_modules/babel-preset-react-app/dependencies.js")
```

**Comment by lucaslcode** [comment]
Nevermind, deleting `node_modules` solved it.

**Pull request merged**